src/Init/Data/List/Basic.lean

```diff
@@ -205,7 +225,15 @@ instance decidableLT [DecidableEq α] [LT α] [DecidableLT α] (l₁ l₂ : List
 @[deprecated decidableLT (since := "2024-12-13"), inherit_doc decidableLT]
 abbrev hasDecidableLt := @decidableLT
 
-/-- The lexicographic order on lists. -/
+/--
+Lexicographic ordering of lists with respect to an ordering on their elements.
+
+`as ≤ bs` if
+ * `as` is empty, or
+ * both `as` and `bs` are non-empty, and the head of `as` is less than the head of `bs`, or
+ * both `as` and `bs` are non-empty, their heads are equal, and the tail of `as` is less than or
+   equal to the tail of `bs`.
```

Comment:
We should be a little careful here. This characterisation is only correct if `[LT α]` satisfies various additional hypotheses.

Maybe say "For a well-behaved `[LT α]` (irreflexive, asymmetric, and antisymmetric; see `Listcons_le_cons_iff` for the precise formulation), `as ≤ bs` if"

Reply:
As I understand it, this is a case of GIGO. The text above still applies for a bogus `LT` instance, it's just that the resulting order is not well-behaved. Right?

So what about instead having an additional sentence like this?

 > This relation is only well-behaved if the underlying `LT α` instance is well-behaved: specifically, it should be irreflexive, asymmetric, and antisymmetric. These requirements are precisely formulated in `List.cons_le_cons_iff`.

Reply:
No, the statement is actually untrue for a bogus LT instance, because le is defined as not lt, and so it doesn't actually need to be related to the lexicographic order coming from the original le. I can come up with a concrete example if needed.

Reply:
Nope, that's sufficient. Thanks.

Reply:
I think this works:
```
Non-strict ordering of lists with respect to a strict ordering of their elements.

`as ≤ bs` if `¬ bs < as`.

This relation can be treated as a lexicographic order if the underlying `LT α` instance is
well-behaved. In particular, it should be irreflexive, asymmetric, and antisymmetric. These
requirements are precisely formulated in `List.cons_le_cons_iff`. If these hold, then `as ≤ bs` if
and only if:
 * `as` is empty, or
 * both `as` and `bs` are non-empty, and the head of `as` is less than the head of `bs`, or
 * both `as` and `bs` are non-empty, their heads are equal, and the tail of `as` is less than or
   equal to the tail of `bs`.
```